interface/reports/appointments_report.php

```diff
@@ -425,22 +490,51 @@ function refreshme() {
     $apptdate_list = array(); // same as above for the appt details
     $totalAppontments = count($appointments);
 
+    $cntr = 1; // column labels above start at 1
     foreach ($appointments as $appointment) {
+        $cntr++;
         array_push($pid_list, $appointment['pid']);
         array_push($apptdate_list, $appointment['pc_eventDate']);
         $patient_id = $appointment['pid'];
         $docname  = $appointment['ulname'] . ', ' . $appointment['ufname'] . ' ' . $appointment['umname'];
 
         $errmsg  = "";
         $pc_apptstatus = $appointment['pc_apptstatus'];
+        if ($_POST['form_csvexport']) {
+            if (empty($appointment['phone_home'])) {
+                $phone_home_csv = $appointment['phone_cell'];
+            } else {
+                $phone_home_csv = $appointment['phone_home'];
+            }
 
-        ?>
+            $name_csv = $appointment['fname'] . " " . substr($appointment['lname'], 0, 1);
+            $dob_csv = $appointment['DOB'];
+            $date_csv = $appointment['pc_eventDate'];
+            $time_csv = $appointment['pc_startTime'];
+
+            $spreadsheet->setActiveSheetIndex(0)
+                ->setCellValue('A' . $cntr, $name_csv)
+                ->setCellValue('B' . $cntr, $dob_csv)
+                ->setCellValue('C' . $cntr, $phone_home_csv)
+                ->setCellValue('D' . $cntr, $date_csv)
+                ->setCellValue('E' . $cntr, $time_csv);
```

Comment:
would escape the _csv values with csvEscape() to prevent csv/formula injection.

also, should label be 'export to XLS' or is this still a CSV?

Reply:
very neat to see this PhpOffice package get used!

Reply:
thanks, it's actually an `XLS` but like @mdsupport was saying this can now be immediately opened with open office calc with no extra clicks

the thing with csvEscape() is that it loads the values into the cells with quotes around them but maybe that's the whole point :laughing:

Reply:
rec making an option/parameter for that function for something like 'bool surroundQuotes = true', which controls if those quotes are added (note in the function that was not really that necessary but done for stray commas). then a call `csvEscape($dob_csv, false)` should hopefully work